core/sbt-cloudflow/src/main/scala/cloudflow/sbt/CloudflowFlinkPlugin.scala

```diff
@@ -0,0 +1,240 @@
+import JsonUtils._
+import cloudflow.sbt.CloudflowKeys._
+import cloudflow.blueprint._
+import cloudflow.blueprint.StreamletDescriptorFormat._
+import cloudflow.sbt.CloudflowKeys._
+import cloudflow.blueprint.StreamletDescriptor
+
+object CloudflowFlinkPlugin extends AutoPlugin {
+  val AppHome = "${app_home}"
+  val AppTargetDir: String = "/app"
+  val appTargetSubdir: String ⇒ String = dir ⇒ s"$AppTargetDir/$dir"
+  val AppJarsDir: String = "app-jars"
+  val DepJarsDir: String = "dep-jars"
+  val optAppDir = "/opt/cloudflow/"
+  final val CloudflowFlinkDockerBaseImage = "lightbend/flink:cloudflow-flink-1.9.2-scala-2.12"
+
+  override def requires = CommonSettingsAndTasksPlugin && StreamletScannerPlugin &&
+    JavaAppPackaging && sbtdocker.DockerPlugin
+
+  override def projectSettings = Seq(
+    libraryDependencies ++= Vector(
+      "com.lightbend.cloudflow" % "cloudflow-runner" % BuildInfo.version,
+      "com.lightbend.cloudflow" %% "cloudflow-flink" % BuildInfo.version,
+      "com.lightbend.cloudflow" %% "cloudflow-flink-testkit" % BuildInfo.version % "test"
+    ),
+    cloudflowDockerParentImage := CloudflowFlinkDockerBaseImage,
+
+    cloudflowFlinkDockerImageName := Def.task {
+      Some(DockerImageName((ThisProject / name).value.toLowerCase, (ThisProject / cloudflowBuildNumber).value.buildNumber))
+    }.value,
+
+    streamletsInProject := Def.taskDyn {
+      val detectedStreamlets = cloudflowStreamletDescriptors.value
+      val dockerImageName = cloudflowFlinkDockerImageName.value
+      buildStreamletDescriptors(detectedStreamlets, dockerImageName)
+    }.value,
+
+    buildOptions in docker := BuildOptions(
+      cache = true,
+      removeIntermediateContainers = BuildOptions.Remove.OnSuccess,
+      pullBaseImage = BuildOptions.Pull.IfMissing
+    ),
+
+    cloudflowStageAppJars := Def.taskDyn {
+      Def.task {
+        val stagingDir = stage.value
+        val projectJars = (Runtime / internalDependencyAsJars).value.map(_.data)
+        val depJars = (Runtime / externalDependencyClasspath).value.map(_.data)
+
+        val appJarDir = new File(stagingDir, AppJarsDir)
+        val depJarDir = new File(stagingDir, DepJarsDir)
+        projectJars.foreach { jar ⇒
+          IO.copyFile(jar, new File(appJarDir, jar.getName))
+        }
+        depJars.foreach { jar ⇒
+          if (jar.name startsWith "cloudflow-runner-") {
+            IO.copyFile(jar, new File(depJarDir, "cloudflow-runner.jar"))
+          } else IO.copyFile(jar, new File(depJarDir, jar.getName))
+        }
+      }
+    }.value,
+
+    imageNames in docker := {
+      // NOTE: only use the default repository name ("lightbend") if a registry
+      //       has been set but no repository has been set.
+      //
+      //       When using the Openshift docker registry, the repository MUST
+      //       already exist as a K8s namespace for the fallback to work.
+      //       We fall back to "lightbend" because we know that Cloudflow will
```

Comment:
This is not true anymore, it is installed in "cloudflow".

Reply:
actually, worse, it's installed where the version configmap is located.

Reply:
yeah I think we need to think about removing this and see if there is a downside to it.

Reply:
deleted ..